cli/azd/docs/external-prompting.md

```diff
@@ -0,0 +1,114 @@
+
+In both cases it would be ideal if `azd` could delegate the prompting behavior back to the caller.  This document outlines a solution, which provides a way for an external tool to provide a remote service that `azd` interacts with when it needs to prompt the user for information.
+
+## Solution
+
+Similar to our strategy for delegating authentication to an external host, we support delegating prompting to an external host via a special JSON based REST API, hosted over a local HTTP server.  When run, `azd` looks for two special environment variables:
+
+- `AZD_UI_PROMPT_ENDPOINT`
+- `AZD_UI_PROMPT_KEY`
+
+When both are set, instead of prompting using the command line - the implementation of our prompting methods now make a POST call to a special endpoint:
+
+`${AZD_UI_PROMPT_ENDPOINT}/prompt?api-version=2024-02-14-preview`
+
+Setting the following headers:
+
+- `Content-Type: application/json`
+- `Authorization: Bearer ${AZD_UI_PROMPT_KEY}`
+
+The use of `AZD_UI_PROMPT_KEY` allows the host to block requests coming from other clients on the same machine (since the it is expected the host runs a ephemeral HTTP server listing on `127.0.0.1` on a random port). It is expected that the host will generate a random string and use this as a shared key for the lifetime of an `azd` invocation.
+
+The body of the request contains information about the prompt that `azd` needs a response for:
+
+```jsonc
+{
+    "type": "<string>", // one of "string", "password", "select", "multiSelect", "confirm",
+    "options": {
+      "message": "<string>", // the message to be displayed as part of the prompt
+      "help": "<string>", // optional help text that can be displayed upon request
+      "options": [ "<string>" /* ... */  ], // for select and multiSelect types - the valid options the user should pick from
+      "defaultValue": "<string>", // optional default value, when undefined there is no default.
+    }
+}
+```
+
+The `password` type represents a string value which represents a password. The host may want to use a different UI element (perhaps one that uses `***` instead of characters) when prompting.
+
+The server should respond with 200 OK and the body should be one of the of the following three shapes:
+
+### Success 
+
+When the host is able to prompt the value it is returned with a `result` of `success`:
+
+```jsonc
+{
+    "result": "success", 
+    "value": "<string> | <string>[]"
+}
+```
+
+When the type is `confirm` the value should be either `"true"` or `"false"` (a string, not a JSON boolean) indicating if the user confirmed the operation (`"true"`) or rejected it (`"false"`). Note that a user rejecting a confirm prompt still results in a `"success"` result (the value is simply `"false"`). In the case of `multiSelect` an array of string values is returned, each individual value is a value from the `"options"` array that was selected by the user.
```

Comment:
What's the reasoning behind using `"true"` / `"false"` instead of `true` / `false`?

Reply:
Yeah - that was my thinking - also that numbers would be returned back as strings instead of JSON numbers.  Happy to change this - just felt like it might be simpler if folks could always expect a string vs an `any` type here - but I am not wed to this decision.